resource_pool/resource_pool_client.py

```diff
@@ -0,0 +1,97 @@
+from contextlib import asynccontextmanager
+from datetime import timedelta
+from typing import AsyncGenerator, Optional
+
+from temporalio import workflow
+
+from resource_pool.shared import (
+    AcquiredResource,
+    AcquireRequest,
+    AcquireResponse,
+    DetachedResource,
+)
+
+
+# Use this class in workflow code that that needs to run on locked resources.
+class ResourcePoolClient:
+    def __init__(self, pool_workflow_id: str) -> None:
+        self.pool_workflow_id = pool_workflow_id
+        self.acquired_resources: list[AcquiredResource] = []
+
+    async def send_acquire_signal(self) -> None:
+        handle = workflow.get_external_workflow_handle(self.pool_workflow_id)
```

Comment:
```suggestion
        handle = workflow.get_external_workflow_handle_for(ResourcePoolWorkflow.run, self.pool_workflow_id)
```

To be more type safe

Reply:
Out of curiosity, this can't help the act of sending the signal, right? It's just there as a best practice for copy/pasters?

Reply:
It actually can in a neat way. When you use `handle.signal(ResourcePoolWorkflow.acquire_resource` below, that does a type check not only on the argument type, but also that that signal is even on the same class. So if you had `handle.signal(SomeOtherWorkflow.acquire_resource` it would fail type check since the generic on the external signal handle is the "self type".

Reply:
Oh that's neat. I didn't look deeply enough to find the wiring that enforces what you described.